contracts/Job.sol

```diff
@@ -37,7 +37,41 @@ contract Job {
 
     function completeJob(uint8 v, bytes32 r, bytes32 s) public returns (bool) {
         require(state == JobState.FUNDED);
-        require(ecrecover(keccak256("\x19Ethereum Signed Message:\n32", keccak256(this)), v, r, s) == consumer);
+
+        bytes32[16] memory conv = [
+            bytes32("000102030405060708090a0b0c0d0e0f"),
+            bytes32("101112131415161718191a1b1c1d1e1f"),
+            bytes32("202122232425262728292a2b2c2d2e2f"),
+            bytes32("303132333435363738393a3b3c3d3e3f"),
+            bytes32("404142434445464748494a4b4c4d4e4f"),
+            bytes32("505152535455565758595a5b5c5d5e5f"),
+            bytes32("606162636465666768696a6b6c6d6e6f"),
+            bytes32("707172737475767778797a7b7c7d7e7f"),
+            bytes32("808182838485868788898a8b8c8d8e8f"),
+            bytes32("909192939495969798999a9b9c9d9e9f"),
+            bytes32("a0a1a2a3a4a5a6a7a8a9aaabacadaeaf"),
+            bytes32("b0b1b2b3b4b5b6b7b8b9babbbcbdbebf"),
+            bytes32("c0c1c2c3c4c5c6c7c8c9cacbcccdcecf"),
+            bytes32("d0d1d2d3d4d5d6d7d8d9dadbdcdddedf"),
+            bytes32("e0e1e2e3e4e5e6e7e8e9eaebecedeeef"),
+            bytes32("f0f1f2f3f4f5f6f7f8f9fafbfcfdfeff")
+        ];
+
+        bytes memory jobAddressHexAsciiBytes = new bytes(42);
+        bytes32 jobAddressBytes = bytes32(uint256(this));
+
+        for (uint i = 0; i < 20; i++) {
```

Comment:
comment describing the algo here including the indexing into the array and the use of the lookup table